assets/entrypoint.sh

```diff
@@ -0,0 +1,228 @@
+#!/bin/bash -e
+# =====================================================================
+# Build script running OpenNMS in Docker environment
+#
+# Source: https://github.com/opennms-forge/docker-horizon-core-web
+# Web: https://www.opennms.org
+#
+# =====================================================================
+
+# Cause false/positives
+# shellcheck disable=SC2086
+
+OPENNMS_HOME="/opt/opennms"
+OPENNMS_OVERLAY="/opt/opennms-overlay"
+
+# Error codes
+E_ILLEGAL_ARGS=126
+E_INIT_CONFIG=127
+E_DEPRECATED_CONFIG=128
+
+# Help function used in error messages and -h option
+usage() {
+  echo ""
+  echo "Docker entry script for OpenNMS service container"
+  echo ""
+  echo "Overlay Config file:"
+  echo "If you want to overwrite the default configuration with your custom config, you can use an overlay"
+  echo "folder in which needs to be mounted to ${OPENNMS_OVERLAY}."
+  echo "Every file in this folder is overwriting files in ${OPENNMS_HOME}."
+  echo ""
+  echo "-h: Show this help."
+  echo "-i: Initialize or update the database and OpenNMS configuration files, runs install -dis."
+  echo "    OpenNMS will be *NOT* started"
+  echo "-s: Start OpenNMS in foreground with an existing configuration."
+  echo "-t: options: Run the config-tester, default is -h to show usage."
+  echo ""
+}
+
+install() {
+  echo "Run OpenNMS install command to initialize or upgrade the database schema and configurations."
+  ${JAVA_HOME}/bin/java -Dopennms.home="${OPENNMS_HOME}" -Dlog4j.configurationFile="${OPENNMS_HOME}"/etc/log4j2-tools.xml -cp "${OPENNMS_HOME}/lib/opennms_bootstrap.jar" org.opennms.bootstrap.InstallerBootstrap "${@}" || exit ${E_INIT_CONFIG}
+}
+
+configTester() {
+  echo "Run config tester to validate existing configuration files."
+  ${JAVA_HOME}/bin/java -Dopennms.manager.class="org.opennms.netmgt.config.tester.ConfigTester" -Dopennms.home="${OPENNMS_HOME}" -Dlog4j.configurationFile="$OPENNMS_HOME"/etc/log4j2-tools.xml -jar $OPENNMS_HOME/lib/opennms_bootstrap.jar "${@}" || exit ${E_INIT_CONFIG}
+}
+
+processConfdTemplates() {
+  echo "Processing confd templates using the backend ${CONFD_BACKEND}."
+  confd -onetime -backend "${CONFD_BACKEND}"
```

Comment:
If a different backend is specified, what happens with the required additional settings like the physical destination for the external server (like etcd, zookeeper, etc.) ?
For example:
```
confd -onetime -backend etcd -node http://127.0.0.1:2379
```
Some backends require more settings.